### PR TITLE
Improved ubuntu build scripts

### PR DESCRIPTION
**Pull request by cyberang3l**
Pull request related to issue #353.
Detailed in information about the pull request in the issue #353, as well as the commit messages.

### CODE REVIEW

File: build_scripts/ubuntu_sim_ros_gazebo.sh [hunk 1]
```diff
@@ -114,13 +117,18 @@ rosinstall_generator mavlink | tee -a /tmp/mavros.rosinstall
 ### Setup workspace & install deps
 wstool merge -t src /tmp/mavros.rosinstall
 wstool update -t src
-rosdep install --from-paths src --ignore-src --rosdistro kinetic -y
+if ! rosdep install --from-paths src --ignore-src --rosdistro kinetic -y; then
```
Comment (hamishwillee) [inline build_scripts/ubuntu_sim_ros_gazebo.sh:120]: @TSC21 Can you please review/sanity check the ROS/Gazebo changes ?

**Comment by hamishwillee** [comment]
@TSC21 can you check the gazebo bits. @dagar Can you confirm none of this will prevent you using these scripts in our docker setup?

@cyberang3l  This all looks better to me thanks! Two things:
1. Can you please add git as a dependency - that is missing on vanilla Ubunut too :-0
2. Can you please do a quick check that the dependencies are still correct in the docs as well following your changes - https://dev.px4.io/en/setup/dev_env_linux_ubuntu.html - in particular the common dependencies.

**Comment by cyberang3l** [comment]
@hamishwillee it seems that git is installed as a dependency by all the build scripts:

```bash
$ git grep apt-get | grep git
ubuntu_sim.sh:43:sudo apt-get install python-argparse git git-core zip python-empy python-toml python-numpy qtcreator cmake build-essential genromfs -y
ubuntu_sim_nuttx.sh:43:sudo apt-get install python-argparse git git-core zip python-empy python-toml python-numpy qtcreator cmake build-essential genromfs -y
ubuntu_sim_ros_gazebo.sh:43:sudo apt-get install python-argparse git git-core zip python-empy python-toml python-numpy qtcreator cmake build-essential genromfs -y
windows_bash_nuttx.sh:40:sudo apt-get install python-argparse git git-core wget zip python-empy python-toml python-numpy cmake build-essential genromfs -y
```

I'll check the docs now.

**Comment by cyberang3l** [comment]
@hamishwillee The English docs have now been updated to reflect these changes!

**Comment by cyberang3l** [comment]
After looking at the scripts there is one more improvement that seems to me logical to implement:

All the scripts are installing at least the common dependencies, but the shell code that installs the common dependencies is copied in each one of the scripts.

If the common dependencies change, then one has to go to each one of these scripts to do the changes (error prone task). It seems logical to me that the `ubuntu_sim_ros_gazebo.sh` and `ubuntu_sim_nuttx.sh` scripts should depend on `ubuntu_sim.sh` (the common dependencies), i.e. when a user downloads e.g. the script `ubuntu_sim_ros_gazebo.sh`, the first thing this script should do is to download and execute the script `ubuntu_sim.sh` before installing the specific ROS/Gazebo stuff.

But if you agree, this is something I can do as a different pull request once this pull request gets merged (in order to avoid very big changes in a single PR).

**Comment by hamishwillee** [comment]
- Thanks for the docs updates. 
- Thanks for checking `git` - thought I hadn't sorted that out yet. 

>  If the common dependencies change, then one has to go to each one of these scripts to do the changes (error prone task). It seems logical to me that the ubuntu_sim_ros_gazebo.sh and ubuntu_sim_nuttx.sh scripts should depend on ubuntu_sim.sh (the common dependencies), i.e. when a user downloads e.g. the script ubuntu_sim_ros_gazebo.sh, the first thing this script should do is to download and execute the script ubuntu_sim.sh before installing the specific ROS/Gazebo stuff.

This was my original intention, but it never occurred to me I could just download dependency scripts :-). Everything needs the common dependencies and Ninja and as you note the scripts are duplicated for that section. 

So I say +1 to this suggestion. I would like @dagar to have chance to comment though, because in the future we hope to use these scripts in our docker config (to allow us to test them and maintain them better) and he owns that task.


The common dependencies and ninja are needed by everything.

File: en/setup/dev_env_linux_ubuntu.md [hunk 2]
```diff
@@ -96,14 +98,16 @@ pip install pyulog
...
 
 ```sh
-wget http://www.eprosima.com/index.php/component/ars/repository/eprosima-fast-rtps/eprosima-fast-rtps-1-5-0/eprosima_fastrtps-1-5-0-linux-tar-gz
-mv eprosima_fastrtps-1-5-0-linux-tar-gz eprosima_fastrtps-1-5-0-linux.tar.gz
+wget http://www.eprosima.com/index.php/component/ars/repository/eprosima-fast-rtps/eprosima-fast-rtps-1-5-0/eprosima_fastrtps-1-5-0-linux-tar-gz -O eprosima_fastrtps-1-5-0-linux.tar.gz
 tar -xzf eprosima_fastrtps-1-5-0-linux.tar.gz eProsima_FastRTPS-1.5.0-Linux/
 tar -xzf eprosima_fastrtps-1-5-0-linux.tar.gz requiredcomponents
 tar -xzf requiredcomponents/eProsima_FastCDR-1.0.7-Linux.tar.gz
-cd eProsima_FastCDR-1.0.7-Linux; ./configure --libdir=/usr/lib; make; sudo make install
+cpucores=$(( $(lscpu | grep Core.*per.*socket | awk -F: '{print $2}') * $(lscpu | grep Socket\(s\) | awk -F: '{print $2}') ))
```
Comment (hamishwillee) [inline en/setup/dev_env_linux_ubuntu.md]: This is a nice feature for  the scripts, and I wouldn't remove it. However "arguably" it is more complicated than needed for people copy-pasting from docs.

Comment (cyberang3l) [reply en/setup/dev_env_linux_ubuntu.md]: I agree. Updated this part.

**Comment by hamishwillee** [comment]
@TSC21 @dagar  Can I please have a sanity check from you on these changes?

**Review by TSC21 — APPROVED** [review]
All good to me @hamishwillee

**Comment by hamishwillee** [comment]
Thanks @TSC21 .  I tested a couple, and they work well.  Merging now.  @cyberang3l  Thanks - looking forward to seeing your improved version :-)

**Pull request merged**